app/org/sagebionetworks/bridge/services/UploadServiceImpl.java

```diff
@@ -82,7 +85,17 @@ public UploadSession createUpload(User user, UploadRequest uploadRequest) {
 
     @Override
     public void uploadComplete(final String uploadId) {
-        final String objectId = uploadDao.getObjectId(uploadId);
+        if (Strings.isNullOrEmpty(uploadId)) {
+            throw new BadRequestException(String.format(Validate.CANNOT_BE_BLANK, "uploadId"));
+        }
+
+        // can't re-complete uploads that are already complete
+        Upload upload = uploadDao.getUpload(uploadId);
+        if (upload.isComplete()) {
+            throw new BadRequestException(String.format("Upload ID %s is already complete", uploadId));
```

Comment:
Not sure about the value of an exception here.  This boolean flag may be set by a worker, for example.  This worker is an internal backend worker, not the worker for unpacking and exporting the uploads by the way.  If the client for some reason never called uploadComplete() but did finish the upload with S3.  The worker may pick up the discrepancy and update the upload table accordingly.